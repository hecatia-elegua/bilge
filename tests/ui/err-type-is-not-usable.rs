#![feature(const_trait_impl)] use bilge::prelude::*;

#[bitsize(2)]
#[derive(TryFromBits)]
enum Dna {
    C,
    G,
    A,
}

fn main() {
    let res = Dna::try_from(u2::new(3));
    match res {
        Ok(Dna::C) => println!("cytosine"),
        Ok(Dna::G) => println!("guanine"),
        Ok(Dna::A) => println!("adenine"),
        Err(bilge::BitsError) => println!("thymine"),
    }
}
