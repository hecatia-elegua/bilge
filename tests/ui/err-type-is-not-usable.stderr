error[E0603]: unit struct `BitsError` is private
  --> tests/ui/err-type-is-not-usable.rs:17:20
   |
17 |         Err(bilge::BitsError) => println!("thymine"),
   |                    ^^^^^^^^^ private unit struct
   |
note: the unit struct `BitsError` is defined here
  --> src/lib.rs
   |
   | pub struct BitsError;
   | ^^^^^^^^^^^^^^^^^^^^
